zcash_client_backend/src/data_api/wallet/input_selection.rs

```diff
@@ -197,6 +202,43 @@ pub trait InputSelector {
     where
         ParamsT: consensus::Parameters,
         ChangeT: ChangeStrategy<MetaSource = Self::InputSource>;
+
+    /// Performs input selection and returns a proposal for the construction of a transaction
+    /// that sends the maximum amount possible from a given account to the specified recipient
+    /// ignoring notes that are below MARGINAL_FEE amount. This transaction will use all the
+    /// funds available minus the resulting fees that will vary according to ZIP-317 specifications.
```

Comment:
The use of ZIP 317 is not determined at this layer; the fees are computed by the `change_strategy`.